packages/contracts/hooks/useEstimateContractFee.tsx

```diff
@@ -32,32 +31,47 @@ export const useEstimateContractFee = <
 >({
   contract,
   functionName,
-  chain,
   executorAddress,
 }: {
   contract: T;
   functionName: FName;
 } & UseEstimateContractFeeProps) => {
+  const { chain } = useWallet();
   const [gasAmountEstimate, setGasAmountEstimate] = useState<bigint | null>(null);
   const [gasPrice, setGasPrice] = useState<bigint | null>(null);
   const [executed, setExecuted] = useState<boolean>(false);
   const [error, setError] = useState<Error | null>(null);
 
-  const publicClient = useMemo(() => initPublicClient(chains[chain]), [chain]);
-  const address = useMemo(() => addresses[contract][chain], [contract, chain]);
-  const abi = useMemo(() => Contracts[contract], [contract]);
+  const publicClient = useMemo(() => (chain ? initPublicClient(chain) : null), [chain]);
```

Comment:
Should we check if it is a valid chain here too?

Reply:
if its not null its valid, chain is a complex object. We check chainId is valid because its just a number